uui/components/datePickers/__tests__/DatePicker.test.tsx

```diff
@@ -103,4 +109,42 @@ describe('DatePicker', () => {
         expect(mocks.onValueChange).toHaveBeenCalledWith('2017-01-31');
         expect(dom.input.value).toEqual('31-01-2017');
     });
+
+    it('should support entering date from keyboard in default format', async () => {
+        const { dom, mocks } = await setupDatePicker({ value: null, format: DATE_FORMAT_DEFAULT });
+        expect(dom.input.value).toEqual('');
+        fireEvent.change(dom.input, { target: { value: '2017-01-22' } });
+        fireEvent.blur(dom.input);
+        expect(mocks.onValueChange).toHaveBeenCalledWith('2017-01-22');
+        expect(dom.input.value).toEqual('Jan 22, 2017');
+    });
+
+    it('should support entering date from keyboard in custom format', async () => {
+        const { dom, mocks } = await setupDatePicker({ value: null, format: DATE_FORMAT_CUSTOM });
```

Comment:
format: DATE_FORMAT_CUSTOM doesn't affect in which format date should be entered in input. We have a support of some of popular formats which hardcoded in DatePicker. Therefore this tests should work without format prop, let's remove it. And let's also test a few more supported formats, you can merge this test with previous one, because that check the similar behavior.

This comment also related for RangeDatePicker tests

Reply:
Added tests with formats.